### DISCUSSION TITLE
cuda like embed opencl

### DISCUSSION THREAD

**Discussion by MatejMagat305** [vlang/v#17083] [Ideas]
I know that is crazy idea, but I want share because I think that it would be interasting feature ..., so the situation seems like that it will be more and more computing on GPU ..., I have 2 sugestion which have advantages and dis. for human read and of course for implementation and they have sub variant:
1. use new extension `.vcl` (or `.vl `or `.vocl`) which would have syntax of v, 
    a) in compilation it would be translated to string opencl by symbol `@OCL("file")` and then normal as currently vsl.vcl (I assume it is relative simplier to implementation)
    b) use this file as normal function (in editor) and in code use syntax https://github.com/vlang/v/blob/master/doc/docs.md#closures and call` some_name[global][local](args)!` - similar to cuda (I asume that it is harder for implementation)

2. use in .v special syntax instead `pub fn name()` import `"example.of.new.cl" `and use `pub fn (cl)name()!`, a) and b) are similar to 1.

of course it is idea after v1.0 and of course it will have backend depended on vsl.vcl or metal or ...

**Comment by MatejMagat305** [comment]
after 2 years I want add it would be like SYCL (similar)

**Reply by JalonSolov** [reply]
Have you checked the OpenCL bindings in https://github.com/vlang/vsl/tree/main/vcl ?
